libcrux-ml-dsa/src/helper.rs

```diff
@@ -0,0 +1,66 @@
+/// The following macros are defined so that the extraction from Rust to C code
+/// can go through.
+
+#[cfg(eurydice)]
+macro_rules! cloop {
+    (for ($i:ident, $chunk:ident) in $val:ident.$values:ident.chunks_exact($($chunk_size:expr),*).enumerate() $body:block) => {
+        for $i in 0..$val.$values.len() / ($($chunk_size)*) {
+            let $chunk = &$val.$values[$i*($($chunk_size)*) .. $i*($($chunk_size)*)+($($chunk_size)*)];
+            $body
+        }
+    };
+    (for ($i:ident, $chunk:ident) in $val:ident.chunks_exact($($chunk_size:expr),*).enumerate() $body:block) => {
+        for $i in 0..$val.len() / ($($chunk_size)*) {
+            let $chunk = &$val[$i*($($chunk_size)*) .. $i*($($chunk_size)*)+($($chunk_size)*)];
+            $body
+        }
+    };
+    (for ($i:ident, $item:ident) in $val:ident.iter().enumerate() $body:block) => {
+        for $i in 0..$val.len() {
+            let $item = &$val[$i];
+            $body
+        }
+    };
+    (for ($i:ident, $item:ident) in $self:ident.$val:ident.iter().enumerate() $body:block) => {
+        for $i in 0..$self.$val.len() {
+            let $item = &$self.$val[$i];
+            $body
+        }
+    };
+    (for ($i:ident, $item:ident) in $val:ident.into_iter().enumerate() $body:block) => {
+        for $i in 0..$val.len() {
+            let $item = $val[$i];
+            $body
+        }
+    };
+    (for $i:ident in ($start:literal..$end:expr).step_by($step:literal) $body:block) => {
+        for $i in $start..$end / $step {
+            let $i = $i * $step;
+            $body
+        }
+    };
+}
+
+#[cfg(not(eurydice))]
+macro_rules! cloop {
```

Comment:
couldn't these be generalized to

```rust
macro_rules! cloop {
  (for $pattern:pat in $iterator:expr $body:block) => {for $pattern in $interator $block}
```

or something similar? I don't think we need to restate the macro for every case.

Reply:
Got it. Also, I think it makes sense to have the same patterns as in the macro_rules for eurydice, so we know the macros expand in both worlds. Let's leave this as-is.